packages/sqs/test/consumers/SqsPermissionConsumer.spec.ts

```diff
@@ -596,6 +597,69 @@ describe('SqsPermissionConsumer', () => {
     })
   })
 
+  describe('multiple consumers', () => {
+    let diContainer: AwilixContainer<Dependencies>
+    let sqsClient: SQSClient
+
+    let publisher: SqsPermissionPublisher
+    let consumer: SqsPermissionConsumer
+
+    beforeEach(async () => {
+      diContainer = await registerDependencies({
+        permissionConsumer: asFunction((dependencies) => {
+          return new SqsPermissionConsumer(dependencies, {
+            creationConfig: {
+              queue: {
+                QueueName: SqsPermissionConsumer.QUEUE_NAME,
+              },
+            },
+            concurrentConsumersAmount: 10,
+            logMessages: true,
+          })
+        }),
+      })
+      sqsClient = diContainer.cradle.sqsClient
+      publisher = diContainer.cradle.permissionPublisher
+      consumer = diContainer.cradle.permissionConsumer
+
+      await consumer.start()
+
+      const command = new ReceiveMessageCommand({
+        QueueUrl: publisher.queueProps.url,
+      })
+      const reply = await sqsClient.send(command)
+      expect(reply.Messages).toBeUndefined()
+    })
+
+    afterEach(async () => {
+      await diContainer.cradle.awilixManager.executeDispose()
+      await diContainer.dispose()
+    })
+
+    it('process all messages properly', async () => {
+      const messagesAmount = 100
+      const messages: PERMISSIONS_ADD_MESSAGE_TYPE[] = Array.from({ length: messagesAmount }).map(
+        (_, i) => ({
+          id: `${i}`,
+          messageType: 'add',
+          timestamp: new Date().toISOString(),
+        }),
+      )
+
+      await Promise.all(
+        messages.map(async (m) => {
+          await publisher.publish(m)
+          await consumer.handlerSpy.waitForMessageWithId(m.id, 'consumed')
```

Comment:
it would be better to do this after the messages are already consumed, and not while we are still publishing them

Reply:
Updated ✅